processors/base/writers/trial_and_record.py

```diff
@@ -126,7 +126,7 @@ def _write_record(conn, trial, record_id, source_id=None, source_url=None, is_pr
         'trial_id': trial['id'],
         'source_id': source_id or record['source_id'],
         'source_url': source_url or record['source_url'],
-        'is_primary': is_primary or record.get('is_primary'),
```

Comment:
New records [don't have `is_primary` property](https://github.com/opentrials/processors/blob/5c21995f82a0d297618f75e7c598bf83ebfcf446/processors/base/writers/trial_and_record.py#L118) at this point in the code so for them `record.get('is_primary')` returns `None`.

The bug is caused by the fact that `is_primary or None` will return `None` if `is_primary` is `False`. Here is some explanatory console output:

```python

In [1]: val = False or None
In [2]: type(val)
Out[2]: NoneType

In [3]: val = None or False
In [5]: val
Out[5]: False

In [8]: val = False or True
In [9]: val
Out[9]: True
```

As you can deduce from this last example, this will also cause records that are no longer primary (`is_primary=False`) to take on their previous value (`True`).
The value of `is_primary` is something [we calculate](https://github.com/opentrials/processors/blob/5c21995f82a0d297618f75e7c598bf83ebfcf446/processors/base/writers/trial_and_record.py#L45) and it should always have the latest value so I just deleted `record.get('is_primary')`.